### ISSUE TITLE
Should we default creation of a bastion public hostname?

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kops#1235]
Right now you have to add this:

```
  topology:
    masters: public
    nodes: public
    bastion:
      bastionPublicName: bastion.example.com
```

Should we do this by default @kris-nova ?

**Comment by chrislovecnm** [comment]
If I could chime in... Because the bastion is not in DNS, is it easier to find to attack?

**Comment by kris-nova** [comment]
@justinsb we have gone back and forth on this quite a bit - and there has been another couple of issues around it

My stance - if key-based `SSH` over TCP:22 is safe enough for github, its safe enough kops. EG - we should be able to trust our security and have no problems opening the port and having a convention like `bastion.example.com`

### TLDR;

Yes, we should do that by default.

**Comment by kris-nova** [comment]
PR : https://github.com/kubernetes/kops/issues/1235
